cmd/machine-config-daemon/start.go

```diff
@@ -167,6 +169,25 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 		return
 	}
 
+	// Use kubelet kubeconfig file to get the URL to kube-api-server
+	kubeconfig, err := clientcmd.LoadFromFile("/etc/kubernetes/kubeconfig")
+	if err != nil {
+		glog.Errorf("failed to load kubelet kubeconfig: %v", err)
+	}
+	clusterName := kubeconfig.Contexts[kubeconfig.CurrentContext].Cluster
+	apiURL := kubeconfig.Clusters[clusterName].Server
+
+	url, err := url.Parse(apiURL)
+	if err != nil {
+		glog.Errorf("failed to parse api url from kubelet kubeconfig: %v", err)
+	}
+
+	// The kubernetes in-cluster functions don't let you override the apiserver
+	// directly; gotta "pass" it via environment vars.
+	glog.V(0).Infof("overriding kubernetes api to %s", apiURL)
+	os.Setenv("KUBERNETES_SERVICE_HOST", url.Hostname())
```

Comment:
One thing to be aware of is that `setenv()` is unsafe in the presence of threads that may be executing C code.  https://internals.rust-lang.org/t/synchronized-ffi-access-to-posix-environment-variable-functions/15475 is a Rust thread on this; which links to e.g. https://sourceware.org/bugzilla/show_bug.cgi?id=15607

The Go runtime uses a middle ground trick of only calling the C `setenv` if cgo is in use which...I think will happen with us when linking to openssl at least.

So we should (at least eventually) fix the client API to allow overriding these things without `setenv()`.

But...for now it's probably OK, I would think (hope) that we're not running any other active goroutines at this point.